CodePush.m

```diff
@@ -505,7 +506,12 @@ - (void)savePendingUpdate:(NSString *)packageHash
 RCT_EXPORT_METHOD(notifyApplicationReady:(RCTPromiseResolveBlock)resolve
                                 rejecter:(RCTPromiseRejectBlock)reject)
 {
-    [CodePush removePendingUpdate];
+    // We only mark a pending update as succeeded only if it has been booted up
+    // successfully, during which `_isFirstRunAfterUpdate` is set to true.
+    if (_isFirstRunAfterUpdate) {
```

Comment:
It would work under the assumption that nobody will `require` the same `react-native-code-push` module twice in the same app, thereby creating two instances. Doing so would defeat the work we are doing to prevent parallel `sync` requests as well as the work to noOp notifyApplicationReady after the first time. Seems like there's no singleton pattern in JS. But I think it is a valid assumption.